localMatchApp/test/logintest.cpp

```diff
@@ -0,0 +1,83 @@
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+#include <iostream>
+#include <string>
+#include "Login.h"
+using namespace testing;
+
+TEST ( LoginTest, TestPasswordGoodCases ) {
+  Login objectOfTest;
+
+  std::string pass1 = "Robert0";
+  std::string pass2 = "R0i";
+  std::string pass3 = "4123412341234123412RR12341234123r";
+  std::string pass4 = "EWRTWERGDFSGSD3DFGDFGSDFr";
+  std::string pass5 = "wertwertwerw4wertwerRtwert";
+  std::string pass6 = "Il0";
+
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass1 ) );
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass2 ) );
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass3 ) );
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass4 ) );
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass5 ) );
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass6 ) );
+}
+
+TEST ( LoginTest, TestPasswordBadCases ) {
+  Login objectOfTest;
+
+  std::string wrongPass1 = "ROBERTO";
+  std::string wrongPass2 = "roberto";
+  std::string wrongPass3 = "4083470";
+  std::string wrongPass4 = "R0BERTO";
+  std::string wrongPass5 = "sdfgsdfg9sdf";
+  std::string wrongPass6 = "EWRTWERsdfgsd";
+
+  EXPECT_FALSE ( objectOfTest.hasCorrectFormat ( wrongPass1 ) );
+  EXPECT_FALSE ( objectOfTest.hasCorrectFormat ( wrongPass2 ) );
+  EXPECT_FALSE ( objectOfTest.hasCorrectFormat ( wrongPass3 ) );
+  EXPECT_FALSE ( objectOfTest.hasCorrectFormat ( wrongPass4 ) );
+  EXPECT_FALSE ( objectOfTest.hasCorrectFormat ( wrongPass5 ) );
+  EXPECT_FALSE ( objectOfTest.hasCorrectFormat ( wrongPass6 ) );
+}
+
+TEST(LoginTest, TestSignUp){
```

Comment:
provide a test description of what you are trying to verify

Reply:
Done.